src/content/docs/dropins/all/extending.mdx

```diff
@@ -12,50 +12,347 @@ import { Steps } from '@astrojs/starlight/components';
+  const toName = formData.get('toName');
+  const giftMessage = formData.get('giftMessage');
 
-## Extend with third-party components
+  giftOptionsField.setAttribute('loading', 'true');
+  console.log('form data', cartId, fromName, toName, giftMessage);
 
-This tutorial describes the steps needed to extend a drop-in by integrating a fictitious third-party ratings & reviews component in the product details drop-in.
+  const giftMessageInput = {
+    from: fromName,
+    to: toName,
+    message: giftMessage,
+  }
 
-:::note[Performance]
-Be vigilant about how you display and fetch data for your third-party component to ensure you avoid any Cumulative Layout Shifts (CLS) that could significantly affect the site's core web vital scores.
-:::
+  fetchGraphQl(`
+    mutation SET_GIFT_OPTIONS($cartId: String!, $giftMessage: GiftMessageInput!) {
+        setGiftOptionsOnCart(input: {
+          cart_id: $cartId,
+          gift_message: $giftMessage
+          gift_receipt_included: false
+          printed_card_included: false
+        }) {
+					cart {
+						id
+						gift_message {
+							from
+							to
+							message
+						}
+					}
+        }
+      }
+    `,
+    {
+    variables: {
+      cartId,
+      giftMessage: giftMessageInput,
+    },
+  }).then(() => {
+    refreshCart();
+    giftOptionsField.removeAttribute('loading');
+  });
+};
+```
+</Task>
+
+<Task>
+### Extend the data payload for the drop-in
+
+To extend the data payload of a drop-in, you need to add a GraphQL fragment to the `cart` payload by modifying the `build.mjs` script at the root of your storefront project. For our example, the fragment should add the gift message data in the cart. Here's an example of what the GraphQL should look like:
+
+```js title='build.mjs'
+/* eslint-disable import/no-extraneous-dependencies */
+import { overrideGQLOperations } from '@dropins/build-tools/gql-extend.js';
+
+// Extend the cart fragment to include the gift message
+overrideGQLOperations([
+  {
+    // The name of the drop-in to extend
+    npm: '@dropins/storefront-cart',
+    // Additional fields to include in the cart results (gift_message)
+    operations: [
+      `fragment CART_FRAGMENT on Cart {
+        gift_message {
+          from
+          to
+          message
+        }
+      }`
+    ],
+  },
+]);
+```
+
+When you run the install command, the `build.mjs` script will generate a new GraphQL query for the cart drop-in that includes the `gift_message` data.  extend the cart fragment to include the gift message data.
+
+</Task>
+
+<Task>
+
+### Add new data to the payload
+
+Map the updated GraphQL query to the data that the cart events provide to listeners. In this way, the checkout drop-in can access gift message data.
+
+Configure the cart drop-in's initializer to add the new cart data to the existing cart payload. This includes the new data with the payload that the cart events provide to listeners. As an example, here is how it might be configured:
```

Comment:
```suggestion
Configure the cart drop-in's initializer to add the new cart data to the existing cart payload. This is done by defining a transformer function on the CartModel. This function receives the GraphQL data and returns an object that gets merged with the rest of the cart payload. As an example, here is how it might be configured:
```